### DISCUSSION TITLE
MSL Port Usage

### DISCUSSION THREAD

**Discussion by danielb-28** [thliebig/openEMS-Project#58] [Q&A Usage] [answered]
I've started studying FDTD and doing some tests with openEMS recently and I have some basic questions concerning the usage of the MSL Port (since I couldn't find an explanation on the wiki/tutorials).  How should I define the position of the excitation and measurement planes and how far can they be from each other and from the limits (start and stop) of the port? Some simulations I did provided very strange results if I put the excitation plane near the starting "face" of the port (the field would freeze at some time step, using a gaussian pulse excitation). Could I treat the measurement plane position the same way as I treat a calibration/reference plane in real life VNA measurements? And how long should the MSL port be? Could I use just a port to simulate a long MSL line or should I use a small port object connected to the actual MSL line (formed by a PEC material)? Are they both equivalent?

I would be very grateful if someone could direct me to some books or materials that would help me to understand those questions better.

**Comment by LubomirJagos** [comment, marked as answer]
These lines are from AddMSLPort.m

```matlab
% create the metal/material for the MSL
MSL_start = start;
MSL_stop = stop;
MSL_stop(idx_height) = MSL_start(idx_height);
CSX = AddBox( CSX, materialname, prio, MSL_start, MSL_stop );
...
port.excite = 0;
if excite
    port.excite = 1;
    CSX = AddExcitation( CSX, [PortNamePrefix 'port_excite_' num2str(portnr)], 0, evec, excite_args{:} );
    CSX = AddBox( CSX, [PortNamePrefix 'port_excite_' num2str(portnr)], prio, ex_start, ex_stop );
end
```

```Q: Could I use just a port to simulate a long MSL line or should I use a small port object connected to the actual MSL line (formed by a PEC material)? Are they both equivalent?```
A: They are both equivalent since AddMSLPort generate box with material (but you have to draw ground plane yourself), it uses excitation box, so you can get equivalent result when draw microstrip by yourself and put AddLumpedPort at beginning.

```Q: And how long should the MSL port be?```
A: As long as you want, you can use it to model whole microstrip line.
